yabt/builders/golang.py

```diff
@@ -191,43 +225,52 @@ def go_builder_internal(build_context, target, command):
     # GOPATH but if it come handy we should implement looking into the docker
     gopaths.append(user_gopath if user_gopath else '/go')
     build_cmd_env = {
-        'XDG_CACHE_HOME': '/tmp/.cache',
+        'XDG_CACHE_HOME': download_cache_dir,
     }
     build_cmd_env.update(target.props.cmd_env or {})
     build_cmd_env['GOPATH'] = ':'.join(gopaths)
 
+    go_mod_path = join(workspace_dir, 'go.mod')
+    user_mod_path = \
+        target.props.get('mod_file', None) or \
+        build_context.conf.get('go_mod_file', None)
+    if user_mod_path:
+        user_mod_path = join(build_context.conf.project_root, user_mod_path)
+        if not isfile(user_mod_path):
```

Comment:
the test is a bit too aggressive caching - last week I help Rotem debug a problem caused by this (switching branches left her with old and bad mod file due to change in proto files)

OK to just leave TODO to test performance and consider regarding the file always

Reply:
I think that go.mod should not be cached. it doesn't improve the performance. It will not download again the external packages even if it is newly created. As long as yabtwork/go/pkg is not removed. It might make it download a new version of an external module.